File: pythonProjects/logica/04_numero_primo.py
import os
import math


def limpiar_consola():
    os.system("cls")


def numero_primo(numero):
    limpiar_consola()
    if numero < 1:
        print(f"{numero} no es un número primo")
    else:
        for i in range(
            2, int(math.sqrt(numero)) + 1
        ):  # Convertir math.sqrt(n) a entero
            if numero % i == 0:
                print(f"{numero} no es un número primo")
                return  # Salir de la función si encontramos un divisor
        print(f"{numero} es un número primo")


if __name__ == "__main__":
    verificar_numero = int(input("Ingrese un valor: "))
    numero_primo(verificar_numero)

File: cppProjects/logica/04_numeroPrimo.cpp
#include <iostream>
#include <cmath>

using namespace std;

bool numeroPrimo()
{
    int numeroUsuario;
    bool esPrimo = true; // Inicialmente asumimos que el número es primo

    cout << "Ingrese un numero: ";
    cin >> numeroUsuario;

    for (int i = 2; i <= sqrt(numeroUsuario); ++i)
    {
        if (numeroUsuario % i == 0)
        {
            esPrimo = false; // Si encontramos un divisor, el número no es primo
            break;           // Salimos del bucle ya que ya no necesitamos comprobar más divisores
        }
    }

    return esPrimo; // Devolvemos el resultado
}

int main()
{
    bool primo = numeroPrimo();

    if (primo)
    {
        cout << "El numero es Primo";
    }
    else
    {
        cout << "El numero no es primo";
    }
    return 0;
}
